JASP-Engine/JASP/R/semsimple.R

```diff
@@ -525,487 +229,646 @@ SEMSimple <- function(dataset=NULL, options, perform="run", callback=function(..
+  semEstimatesTable$addColumnInfo(name="std.nox", title = "std (nox)", type="number")
+  semEstimatesTable$addColumnInfo(name="group",  title = "group", type="string")
+  
+  semEstimatesTable$addCitation("Rosseel, Y. (2012). lavaan: An R Package for Structural Equation Modeling. Journal of Statistical Software, 48(2), 1-36. URL http://www.jstatsoft.org/v48/i02/")
+  semEstimatesTable$showSpecifiedColumnsOnly <- TRUE
+  
+  if (!is.null(semContainer[["semResultsList"]]$object$semResults)) {
+    
+    semEstimates <- lavaan:::parameterEstimates(semContainer[["semResultsList"]]$object$semResults, standardized=TRUE)
+    semEstimatesTable$setData(semEstimates)
+    semContainer[["semEstimatesTable"]] <- semEstimatesTable
+    
+  } 
+  
+  return()
+}
 
-  ### ANOVA table ###
-  an0va <- list()
-  an0va[["title"]] <- "Chi Square Test Statistic (unscaled)"
-  # an0va[["cases"]] <- c("Saturated", "Model")
-  an0va[["schema"]] <- list(
-    fields = list(
-      list(name="Model", title = "", type="string"),
-      list(name="DF", title = "df", type="number", format="dp:0"),
-      list(name="AIC", type="number", format="dp:1"),
-      list(name="BIC", type="number", format="dp:1"),
-      list(name="Chisq", title = "&#967;&sup2;", type="number", format="dp:3"),
-      list(name="Chisq diff", title = "&#916;&#967;&sup2;", type="number", format="dp:3"),
-      list(name="Pr(>Chisq)", title = "p", type="number", format="dp:3;p:0.001")
-    )
-  )
-  an0va[["data"]] <- list()
-  ### PERFORM = RUN
-  if (is.null(semResults)) {
-    sem_anova <- structure(list(Df=c(0, NA), AIC=c(NA, NA), BIC=c(NA, NA),
-                                Chisq = c(0, NA), `Chisq diff`=c(NA, NA),
-                                `Df diff` = c(NA, NA), `Pr(>Chisq)` = c(NA, NA)),
-                           .Names = c("Df", "AIC", "BIC", "Chisq", "Chisq diff", "Df diff", "Pr(>Chisq)"),
-                           row.names = c("Saturated", options$modelName),
-                           class = c("anova", "data.frame"),
-                           heading = "Chi Square Test Statistic (unscaled)\n")
+.semMainTable <- function(semContainer, options, dataset, ready) {
+  if (!ready || !is.null(semContainer[["semFitTable"]]))
+    return()
+  
+  semFitTable <- createJaspTable(title = "Chi Square Test Statistic (unscaled)")
+  
+  semFitTable$addColumnInfo(name="Model", title = "", type="string")
+  semFitTable$addColumnInfo(name="Df", title = "df", type="number")
+  # semFitTable$addColumnInfo(name="DfDiff", title = "&#916;df", type="number")
+  semFitTable$addColumnInfo(name="AIC", type="number")
+  semFitTable$addColumnInfo(name="BIC", type="number")
+  semFitTable$addColumnInfo(name="Chisq", title = "&#967;&sup2;", type="number")
+  # semFitTable$addColumnInfo(name="ChisqDiff", title = "&#916;&#967;&sup2;", type="number")
+  semFitTable$addColumnInfo(name="PrChisq", title = "p", type="number")
```

Comment:
For all p values put `type = "pvalue"`, this ensures they behave properly (showing < 0.001, etc.)